sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -87,23 +105,77 @@ def get_base_images_sbom_components(base_images_digests, is_last_from_scratch):
     return components
 
 
+def get_base_images_from_dockerfile(parsed_dockerfile):
+    """
+    Reads the base images from provided parsed dockerfile
+
+    :param parsed_dockerfile: (Dict) - Contents of the parsed dockerfile
+    :return: base_images (List) - List of base images used during build as extracted
+                                  from the dockerfile in the order they were used.
+
+    Example:
+    If the Dockerfile looks like
+    FROM registry.access.redhat.com/ubi8/ubi:latest as builder
+    ...
+    FROM builder
+    ...
+
+    Then the relevant part of parsed_dockerfile look like
+    {
+        "Stages": [
+            {
+                "BaseName": "registry.access.redhat.com/ubi8/ubi:latest",
+                "As": "builder",
+                "From": {"Image": "registry.access.redhat.com/ubi8/ubi:latest"},
+            },
+            {
+                "BaseName": "builder",
+                "From": {"Stage": {"Named": "builder", "Index": 0}},
+            },
+        ]
+    },
+    """
+    base_images = []
+
+    # this part of the json is the relevant one that contains the
+    # info about base images
+    stages = parsed_dockerfile["Stages"]
+
+    for stage in stages:
+        if "Image" in stage["From"]:
+            base_images.append(stage["From"]["Image"])
+        elif "Scratch" in stage["From"]:
+            base_images.append("scratch")
+        elif "Stage" in stage["From"]:
+            stage_refering_to_index = stage["From"]["Stage"]["Index"]
+            # Find the original stage/image. Named stage can refer to another named stage,
+            # so continue looking until we find the image those stages refer to.
+            while stage_refering_to_index is not None and stage_refering_to_index != {}:
+                refered_stage = stages[stage_refering_to_index]
+                stage_refering_to_index = refered_stage.get("From").get("Stage", {}).get("Index", None)
+                if stage_refering_to_index is None or stage_refering_to_index == {}:
+                    base_images.append(refered_stage["From"]["Image"])
```

Comment:
Is it technically possible to have `FROM scratch AS somestage`? When we find the original stage, we should probably apply the same image-getting logic to it that we apply to every stage

Reply:
Hmm, that perversion is indeed possible. But that brings another issue, for example
```
FROM scratch AS stageA
COPY . .

FROM stageA
COPY . .
```
When parsed, it does not behave well - e.g. the stageA, does not hold a index reference to the original from scratch, but only has "Image", which is not what that is at all, so it is not even parsed "properly" and it would be hard for us to start deciding whether is is just a perversion - reference to FROM stage or an actual image. That would bring another layer of complexity unnecessarily I would say. Nobody should ever use that :D  
```
{
    "Stages": [
        {
            "BaseName": "scratch",
            "As": "stageA",
            "From": {"Scratch": True},
        },
        {
            "BaseName": "stageA",
            "From": {"Image": "stageA"},
        },
    ]
},
```

Reply:
Ack, there probably shouldn't be any legitimate use case for this